### ISSUE TITLE
Why scale the attention mask with (-100)?

### ISSUE THREAD

**Issue by Chanfeechen** [pals-ttic/adapting-CLIP#4]
Hi,

Your work is very interesting! I have a doubt while I look into the implementation of spatial ViT. I found that you scale the attention mask with (-100) when you [convert the mask to attention](https://github.com/pals-ttic/adapting-CLIP/blob/main/spatial_clip/vit/spatial_vit.py#:~:text=attn_map%20%3D%20attn_map.bool().half()%20*%20(%2D100)). Could you help me understand the reason for this scaling? 

Thank you very much!
Peijie

**Comment by raymondyeh07** [comment]
Hi Peijie,

Thank you for your interest in our work. 

Recall, in [MultiheadAttention](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html) if `attn_mask` is provided
> For a float mask, the mask values will be added to the attention weight.

In a transformer, the attention is computed via a SoftMax, adding a very negative number to an element effectively turns its output to zero, i.e, the attention is zero. 

Note, the masks computed from SLIC contains elements in {0,1}. We multiply by (-100) such that the elements that are originally 1 lead to an attention that's effectively zero.

Best,
Raymond

**Comment by Chanfeechen** [comment]
Thank you very much for the explanations! That makes sense!
